### PR TITLE
Add a new dockerfile for worker-mode synapse

### PR DESCRIPTION
**Pull request by anoadragon453**
This PR adds a new Dockerfile containing the configuration for a Synapse running in worker mode. Some additional files were required to support the image.

1. A caddy binary needed to be placed in the container for termination TLS requests on the federation port (thanks to Conduit for the idea!). The included file is a Caddy config json file.
2. A shared config file to hand to each Synapse worker was necessary. This file is subtley different from [dockerfiles/synapse/homeserver.yaml](https://github.com/matrix-org/complement/blob/master/dockerfiles/synapse/homeserver.yaml), though it may be nice to have both files be derived from a shared file in the future.

The intention of this is to add testing of Synapse in worker mode to Complement's CI. It is paired with https://github.com/matrix-org/synapse/pull/9162.

### CODE REVIEW

File: dockerfiles/SynapseWorkers.Dockerfile
Comment (Kegsay) [inline dockerfiles/SynapseWorkers.Dockerfile:26]: Not following this here. Complement will auto-mount a volume to `/ca` which will contain the certs. Why are we doing this copy from a `/keys` directory?

File: dockerfiles/SynapseWorkers.Dockerfile
Comment (Kegsay) [inline dockerfiles/SynapseWorkers.Dockerfile:39]: I'd move this up near the top of the file to make better use of docker caching.

**Review by Kegsay — APPROVED** [review]

File: dockerfiles/SynapseWorkers.Dockerfile
Comment (anoadragon453) [reply dockerfiles/SynapseWorkers.Dockerfile:26]: Good point. This was just copied over from the existing Synapse config, and can go. However I did notice that having `ENV COMPLEMENT_CA=true` in the dockerfile (at least in this setup) doesn't seem to goad Complement into running in CA mode.

Including the variable when running the dockerfile works as expected.

Comment (Kegsay) [reply dockerfiles/SynapseWorkers.Dockerfile]: `COMPLEMENT_CA=true` is a config option for running Complement itself, it's driven by the end-user not by random dockerfiles. It's up to the dockerfile to look in `/ca` if it exists to know whether to use it or not.

Comment (anoadragon453) [reply dockerfiles/SynapseWorkers.Dockerfile]: I see, I was confused by the "or the docker container" bit in the README then: https://github.com/matrix-org/complement#complement-pki.

**Comment by Kegsay** [comment]
Is this ready to merge?

**Comment by anoadragon453** [comment]
Yeah, sorry. I was just waiting on the CI stuff to get sorted before merging. But that might be a while and I want to put some more PRs up for fixing cert stuff in Complement that would require more changes here.

So let's just merge it now and do anything more in other PRs.

**Pull request merged**
